backend/src/sync/mod.rs

```diff
@@ -124,3 +146,40 @@ impl VersionResponse {
         self.version.parse().expect("invalid version string")
     }
 }
+
+
+#[derive(Debug, Serialize)]
+pub enum DeletionMode {
+    Series { eager: bool },
+    Events { eager: bool },
+    Playlists { eager: bool },
+}
+
+impl FromStr for DeletionMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "series" => Ok(Self::Series { eager: false }),
+            "series:eager" => Ok(Self::Series { eager: true }),
+
+            "events" => Ok(Self::Events { eager: false }),
+            "events:eager" => Ok(Self::Events { eager: true }),
+
+            "playlists" => Ok(Self::Playlists { eager: false }),
+            "playlists:eager" => Ok(Self::Playlists { eager: true }),
+
+            other => Err(format!("Invalid auto_delete_pages value: {}", other)),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for DeletionMode {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        DeletionMode::from_str(&s).map_err(serde::de::Error::custom)
+    }
+}
```

Comment:
Uhm can you not replace this with `derive(Deserialize)` and `#[serde(try_from = "String")]`? We do it like that in some other places.